components/script/script_module.rs

```diff
@@ -357,14 +357,14 @@ impl ModuleTree {
         let comp = InRealm::Entered(&realm);
         let _ais = AutoIncumbentScript::new(&owner.global());
 
+        let new_promise = Promise::new_in_current_realm(comp);
+
         let mut promise = self.promise.borrow_mut();
```

Comment:
This borrow is still active when we call append_native_handler, so we need to reduce its scope.

Reply:
Ideally we only use borrow calls in expressions/statements that need them, rather than open-ended ones like this.